primitives/runtime/src/offchain/storage_lock.rs

```diff
@@ -337,6 +355,17 @@ impl<'a, 'b, L: Lockable> StorageLockGuard<'a, 'b, L> {
 	pub fn forget(mut self) {
 		let _ = self.lock.take();
 	}
+
+	/// Extend the lock by guard if the lock is existed.
+	///
+	/// Can be used to extend deadline when doing iterations
+	/// that it's hard to be estimated consumed time. After
+	/// a successful step, extend the lock if the lock is still active.
+	pub fn extend_lock(&mut self) {
```

Comment:
The name `extend` is slightly misleading, since we really restart the deadline "countdown" so to speak, we don't prolong it by a value, but I don't have a better name either.

Reply:
Is `refresh` ok?

Reply:
I think we can leave as is, I did not come up with a better name.